cache/cache.go

```diff
@@ -48,12 +48,22 @@ func NewRandomEvictedCache(maxBytes int) Cache {
 	}
 }
 
+func (c *randomEvictedCache) entrySize(key *string, value Measurable) int {
```

Comment:
Looks like calling it copy-on-write is not strictly accurate, even if the semantics are the same:
https://stackoverflow.com/a/8532749